oauth/src/main/java/com/halcyonmobile/oauth/OauthRetrofitContainerBuilder.kt

```diff
@@ -17,6 +17,7 @@
 package com.halcyonmobile.oauth
 
 import com.halcyonmobile.oauth.dependencies.AuthenticationLocalStorage
+import com.halcyonmobile.oauth.dependencies.IsSessionExpiredException
```

Comment:
This introduces a breaking change in the API.
I think it would be better to keep  the original class also and mark as Deprecated.
Also create another function in this class with the same name `setIsSessionExpiredExceptionDecider` deprecated.
We could import it as `import com.halcyonmobile.oauth.IsSessionExpiredException as DeprecatedIsSessionExpiredException`

Then if the deprecated method is used, we could use an Adapter:
The adapter extends the new IsSessionExpiredException and receives a DeprecatedIsSessionExpiredException as its constructor argument.
in invoke it checks if the throwable is HttpException && the deprecated returns true.